breakpad-symbols/src/sym_file/parser.rs

```diff
@@ -1111,39 +1078,40 @@ FUNC 1001 10 10 some func overlap contained
         assert_eq!(p.parameter_size, 0x3);
         assert_eq!(p.name, "func 2".to_string());
     }
-    assert_eq!(sym.functions.ranges_values().count(), 1);
-    let funcs = sym
-        .functions
-        .ranges_values()
-        .map(|&(_, ref f)| f)
-        .collect::<Vec<_>>();
+    assert_eq!(sym.functions.len(), 2);
+    assert!(sym
+        .function_for_address(0x1000)
+        .unwrap()
+        .name
+        .starts_with("some func"));
+    assert_eq!(sym.function_for_address(0x1030).unwrap().address, 0x1001);
```

Comment:
! what's up with this

Reply:
Yeah this is probably the most major change in behavior. We now allow overlapping functions if they aren't fully subsumed. And for the addresses in the overlapping parts, we don't care which function is returned. In the first test, all of the candidates begin with "some func". And the second one is testing that the function which starts at 0x1001 and sticks out the end from the previous one actually covers the byte at 0x1030.